### PR TITLE
Correct the timestamp used by the camera (kinetic-devel)

### PR DESCRIPTION
**Pull request by j-rivero**
{ port of pull request #410 }

Fix for #408 

The last measurement time is the time that gazebo generated the sensor data, so ought to be used. The other cameras need similar fixes to have the proper timestamps. There is a test to show that the timestamp is good, but issue #409 has lead me to comment it out for now.

There probably ought to be a separate issue about what updateRate is supposed to be for, and if it isn't useful remove it (from all the sensors that have it) in another PR.

### CODE REVIEW

File: gazebo_plugins/src/gazebo_ros_camera.cpp [hunk 1]
```diff
@@ -76,7 +76,11 @@ void GazeboRosCamera::OnNewFrame(const unsigned char *_image,
     unsigned int _width, unsigned int _height, unsigned int _depth,
     const std::string &_format)
 {
-  this->sensor_update_time_ = this->parentSensor_->LastUpdateTime();
+# if GAZEBO_MAJOR_VERSION >= 7
+  common::Time sensor_update_time = this->parentSensor_->LastMeasurementTime();
+# else
+  common::Time sensor_update_time = this->parentSensor_->GetLastMeasurementTime();
```
Comment (davetcoleman) [inline gazebo_plugins/src/gazebo_ros_camera.cpp:82]: Is there a use case when the kinetic-devel branch will be used with a version of Gazebo less than 7? Seems this compiler directive can be removed

File: gazebo_plugins/test/camera/camera.cpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+// Test if the camera image is published at all, and that the timestamp
+// is not too long in the past.
+TEST_F(CameraTest, cameraSubscribeTest)
+{
+  image_transport::ImageTransport it(nh_);
+  cam_sub_ = it.subscribe("camera1/image_raw", 1,
+                          &CameraTest::imageCallback,
+                          dynamic_cast<CameraTest*>(this));
+
+#if 0
```
Comment (davetcoleman) [inline gazebo_plugins/test/camera/camera.cpp:34]: explain when this test should be re-enabled (link to issue) or remove this commented out code

File: gazebo_plugins/CMakeLists.txt [hunk 3]
```diff
@@ -361,11 +363,28 @@ install(DIRECTORY test
...
+    target_link_libraries(depth_camera-test ${catkin_LIBRARIES})
+    add_rostest_gtest(multicamera-test
+                      test/camera/multicamera.test
+                      test/camera/multicamera.cpp)
+    target_link_libraries(multicamera-test ${catkin_LIBRARIES})
+    add_rostest_gtest(camera-test
+                      test/camera/camera.test
+                      test/camera/camera.cpp)
+    target_link_libraries(camera-test ${catkin_LIBRARIES})
+  endif()
```
Comment (davetcoleman) [inline gazebo_plugins/CMakeLists.txt:389]: awesome tests!

**Review by davetcoleman — APPROVED** [review]
See minor comments

**Comment by davetcoleman** [comment]
The build is failing because of a CMake warning from gazebo_ros_control:

```
08:37:08 CMake Warning at /opt/ros/kinetic/share/catkin/cmake/catkin_package.cmake:166 (message):
08:37:08   catkin_package() DEPENDS on 'gazebo' but neither 'gazebo_INCLUDE_DIRS' nor
08:37:08   'gazebo_LIBRARIES' is defined.
08:37:08 Call Stack (most recent call first):
08:37:08   /opt/ros/kinetic/share/catkin/cmake/catkin_package.cmake:102 (_catkin_package)
08:37:08   CMakeLists.txt:22 (catkin_package)
```

That I believe I fixed in my [PR](https://github.com/ros-simulation/gazebo_ros_pkgs/pull/537) :)

File: gazebo_plugins/src/gazebo_ros_camera.cpp (hunk 1 above)
Comment (j-rivero) [reply gazebo_plugins/src/gazebo_ros_camera.cpp:82]: I'm ok with removing support for gazebo < 7 in the kinetic branch but let's do it a single PR and do not spread removals all over the place to keep PRs clean.

**Pull request merged**